cli/lsp/resolver.rs

```diff
@@ -419,64 +429,111 @@ impl LspNpmConfigHash {
     let mut hasher = FastInsecureHasher::new();
     hasher.write_hashable(scope);
     hasher.write_hashable(node_modules_dir);
-    hasher.write_hashable(global_cache_path);
     if let Some(lockfile) = lockfile {
       hasher.write_hashable(&*lockfile.lock());
     }
-    hasher.write_hashable(global_cache_path);
+    hasher.write_hashable(cache.deno_dir().npm_folder_path());
     Self(hasher.finish())
   }
 }
 
+#[derive(Debug)]
+struct RedirectEntry {
+  target: ModuleSpecifier,
+  headers: Arc<HashMap<String, String>>,
+  destination: Option<ModuleSpecifier>,
+}
+
 #[derive(Debug)]
 struct RedirectResolver {
```

Comment:
The changes in here seem a little complicated. Do we have unit tests for this? If not, how hard would it be to add unit tests?

Reply:
Seems like there's no way of constructing a `HttpCacheItemKey` because of its field visibilities, meaning I can't mock `HttpCache` here.

Reply:
It looks like a seam could be created here?

```rs
      let Some(cache_key) = self.cache.cache_item_key(&current).ok() else {
        break None;
      };
      let Some(headers) = self.cache.read_headers(&cache_key).ok().flatten()
      else {
        break None;
      };
```

```rs
let Some(headers) = (self.get_headers)(&current) else {
  break None;
}
```